core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -296,30 +296,35 @@ class ReassignPartitionsCommand(zkUtils: ZkUtils, partitions: Map[TopicAndPartit
   }
 
   def addThrottledReplicaList(): Unit = {
-    //apply the throttle to all move destinations and all move sources
     val existing = zkUtils.getReplicaAssignmentForTopics(partitions.map(_._1.topic).toSeq)
-    val moves = replicaMoves(existing, proposed = partitions)
-    for (topic <- partitions.keySet.map(tp => tp.topic).toSeq.distinct)
-      AdminUtils.changeTopicConfig(zkUtils, topic, new Properties {
-        put(LogConfig.ThrottledReplicasListProp, moves.get(topic).get )
-      })
-    println(s"Throttles were added to the following replicas: $moves")
+    val throttled = topicToLeaderAndFollowerThrottles(existing, partitions)
+    for (topic <- partitions.keySet.map(tp => tp.topic).toSeq.distinct) {
+      val props = new Properties()
+      val (leaderThrottledReplicas, followerThrottledReplicas) = throttled.get(topic).get
+      props.put(LogConfig.LeaderThrottledReplicasListProp, leaderThrottledReplicas)
+      props.put(LogConfig.FollowerThrottledReplicasListProp, followerThrottledReplicas)
+      AdminUtils.changeTopicConfig(zkUtils, topic, props)
+    }
+    println(s"Throttles were added for topics ${throttled.keySet}")
   }
 
-  def replicaMoves(existing: Map[TopicAndPartition, Seq[Int]], proposed: Map[TopicAndPartition, Seq[Int]]): Map[String, String] = {
-    //Find the replicas that have moved
+  def topicToLeaderAndFollowerThrottles(existing: Map[TopicAndPartition, Seq[Int]], proposed: Map[TopicAndPartition, Seq[Int]]): Map[String, (String, String)] = {
+    //Find the replicas that are either move destinations (where we're moving to, aka follower throttles)
+    //or move sources (which are defined as all current replicas - aka leader throttles)
```

Comment:
It seems we are missing the part to subtract after "which are defined as all current replicas - ".